person.rb

```diff
@@ -7,12 +8,17 @@ def initialize(age:, name: 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @corrector = corrector
   end
 
   def can_use_services?
     of_age? || true
   end
 
+  def validate_name
+    @name = @corrector.correct_name(name)
```

Comment:
Nicely done, you used the Corrector instance to validate and save the corrected name. 😎👍

Reply:
Done ✔